runtime_builders/builder_util.py

```diff
@@ -57,10 +59,51 @@ def get_file_from_gcs(gcs_file, temp_file):
         return False
 
 
-def write_manifest_file(manifest):
+def verify_and_write_manifest(manifest):
```

Comment:
Could you split this into a verify method and a write method? Then you could add a set of unit tests for the verify portion.[